python/lsst/ts/externalscripts/auxtel/latiss_cwfs_align.py

```diff
@@ -0,0 +1,547 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["LatissCWFSAlign"]
+
+import os
+import copy
+import time
+import yaml
+import wget
+import asyncio
+import warnings
+import logging
+
+import concurrent.futures
+
+from pathlib import Path
+
+import numpy as np
+from astropy import time as astropytime
+from astropy.io import fits
+
+from scipy import ndimage
+from scipy.signal import medfilt
+from scipy.ndimage.filters import gaussian_filter
+from astropy.modeling import models, fitting
+
+from lsst.ts import salobj
+from lsst.ts.standardscripts.auxtel.attcs import ATTCS
+from lsst.ts.standardscripts.auxtel.latiss import LATISS
+from lsst.ts.idl.enums.Script import ScriptState
+
+import lsst.daf.persistence as dafPersist
+
+# Source detection libraries
+from lsst.meas.algorithms.detection import SourceDetectionTask
+
+# cosmic ray rejection
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask
+
+import lsst.afw.table as afwTable
+
+from operator import itemgetter
+from lsst.ip.isr.isrTask import IsrTask
+
+import matplotlib.pyplot as plt
+
+# Import CWFS package
+from lsst import cwfs
+from lsst.cwfs.instrument import Instrument
+from lsst.cwfs.algorithm import Algorithm
+from lsst.cwfs.image import Image, readFile, aperture2image, showProjection
+import lsst.cwfs.plots as plots
+
+
+class LatissCWFSAlign(salobj.BaseScript):
+    """ Perform an optical alignment procedure of Auxiliary Telescope with
+    the GenericCamera CSC.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    * intra: when taking intra focus image.
+    * extra: when taking extra focus image.
+    * cwfs: when running CWFS code.
+
+    **Details**
+
+    This is what the script does:
+
+    * TBD
+    """
+
+    __test__ = False  # stop pytest from warning that this is not a test
+
+    def __init__(self, index, remotes=True):
+
+        super().__init__(index=index,
+                         descr="Perform optical alignment procedure on LATISS data.")
+
+        self.attcs = None
+        self.latiss = None
+        if remotes:
+            self.attcs = ATTCS(self.domain)
+            self.latiss = LATISS(self.domain)
+
+        self.short_timeout = 5.
+        self.long_timeout = 30.
+
+        # Sensitivity matrix: mm of hexapod motion for nm of wfs. To figure out
+        # the hexapod correction multiply
+        self.sensitivity_matrix = [[-1. / 131., 0., 0.],
```

Comment:
Once we get this established we have to move these into configs in the AOS. Fine for now